keyboards/punk75/keymaps/via/keymap.c

```diff
@@ -0,0 +1,75 @@
+ * |        |        | RGB TG | FN     | RGB RMD| RGB MD | P0     |        | P.     | PENT   | PENT   | FN     |        |        |        |
+ * '--------------------------------------------------------------------------------------------------------------------------------------'
+ */
+
+    [1] = LAYOUT_ortho_5x15( /* FUNCTION */
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_NLCK, KC_SLSH, KC_ASTR, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+        KC_MSEL, KC_CALC, KC_MYCM, KC_MAIL, RGB_HUD, RGB_HUI, KC_P7,   KC_P8,   KC_P9,   KC_MINS, _______, _______, KC_PSCR, KC_SLCK, KC_PAUS,
+        KC_MPRV, KC_MPLY, KC_MNXT, KC_MSTP, RGB_SAD, RGB_SAI, KC_P4,   KC_P5,   KC_P6,   KC_PLUS, _______, RESET,   _______, _______, _______,
+        KC_VOLD, KC_MUTE, KC_VOLU, KC_APP,  RGB_VAD, RGB_VAI, KC_P1,   KC_P2,   KC_P3,   KC_PENT, _______, _______, _______, _______, _______,
+        _______, _______, RGB_TOG, MO(1), RGB_RMOD,RGB_MOD, KC_P0,   _______, KC_PDOT, KC_PENT, KC_PENT, MO(1), _______, _______, _______
+    ),
+
+    [2] = LAYOUT_ortho_5x15(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    ),
+
+    [3] = LAYOUT_ortho_5x15(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    ),
+};
+
+void led_keypress_update(pin_t led_pin, uint16_t keycode, keyrecord_t *record) {
+    // When a key is pressed turn on the LED, when released turn it off
+    writePin(led_pin, record->event.pressed);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    // Update LED state
+    led_keypress_update(LED, keycode, record);
+    return true;
+}
+
```

Comment:
Is the encoder code left out on purpose?

Reply:
I do not know how the encoder code works with VIA, and they are also optional. I'll investigate tomorrow and report back with a change or a merge request_

Reply:
Think of via keymaps as a keymap where some parts of it can be overwritten dynamically by via. Anything that can't should be baked into it like with normal QMK

Encoder code is an example of code that can't be changed with via. What most people commonly do is decide on a standard function for the encoder, most do volume control.